libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -36,11 +36,9 @@ using namespace Linux;
 UARTDriver::UARTDriver(bool default_console) :
     device_path(NULL),
     _packetise(false),
-    _flow_control(FLOW_CONTROL_DISABLE)
+    _device{new ConsoleDevice()}
```

Comment:
It seems like sometimes some methods of UARTDriver are called without ```_device``` allocated (before ```begin()```) which results in SEGFAULTs. I figured, ```ConsoleDevice``` is a good default value since it's been like that before. Checking if ```OwnPtr<>``` owns a nullptr seems a little bit awkward.